### PR TITLE
Ceil function partition added

### LINKED ISSUE #1443: [Feature Request] Automatically generate partition filters for generated columns using the ceiling function
See #1442 

We can automatically generated partition filters when a partition column is generated by the [ceil](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/SupportedGenerationExpressions.scala#L82) function.

i.e. if x < y then ceil(x) <= ceil(y)

### PR DESCRIPTION
**Pull request by sherlockbeard**
## Description

Resolves #1443 
## How was this patch tested?

added test in OptimizeGeneratedColumnSuite

### CODE REVIEW

**Comment by allisonport-db** [comment]
Hi @sherlockbeard sorry for the delay, I will try to look at this later today

**Review by allisonport-db — CHANGES_REQUESTED** [review]
We want to support all data types supported for the [Ceil function](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L251) which also includes DecimalType and LongType

**Comment by sherlockbeard** [comment]
cc @allisonport-db

**Comment by sherlockbeard** [comment]
cc @allisonport-db

File: core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala [hunk 1]
```diff
@@ -20,9 +20,9 @@ import org.apache.spark.sql.delta.OptimizablePartitionExpression._
 
 import org.apache.spark.sql.Column
 import org.apache.spark.sql.catalyst.dsl.expressions._
-import org.apache.spark.sql.catalyst.expressions.{Cast, DateFormatClass, DayOfMonth, Expression, Hour, IsNull, Literal, Month, Or, Substring, UnixTimestamp, Year}
+import org.apache.spark.sql.catalyst.expressions.{Cast, Ceil, DateFormatClass, DayOfMonth, Expression, Hour, IsNull, Literal, Month, Or, Substring, UnixTimestamp, Year}
 import org.apache.spark.sql.catalyst.util.quoteIfNeeded
-import org.apache.spark.sql.types.{DateType, StringType, TimestampType}
+import org.apache.spark.sql.types.{DateType, DoubleType, IntegerType, StringType, TimestampType, LongType, DecimalType}
```
[the remark this answers is not in the document]
Comment (allisonport-db) [reply core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala:25]: Can you check if IntegerType or any other datatypes are implicitly cast?

[the remark this answers is not in the document]
Comment (sherlockbeard) [reply core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala]: sorry had mail sync off so missed this message .
corrected it.

[the remark this answers is not in the document]
Comment (sherlockbeard) [reply core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala]: i added IntegerType import in first commit . after that code was changed but forgot to take it out .

[the remark this answers is not in the document]
Comment (sherlockbeard) [reply core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala]: cc @allisonport-db

**Comment by Nestor10** [comment]
Hello, I was looking for a first issue in delta. Is there anything I could do to get this PR off the floor? This seems fairly complete.

**Pull request closed without merging**